packages/tldraw/src/lib/tools/SelectTool/childStates/Idle.ts

```diff
@@ -499,6 +510,28 @@ export class Idle extends StateNode {
 
 	isDarwin = window.navigator.userAgent.toLowerCase().indexOf('mac') > -1
 
+	isOverArrowLabelTest(shape: TLShape | undefined) {
+		if (!shape) return false
+
+		const pointInShapeSpace = this.editor.getPointInShapeSpace(
+			shape,
+			this.editor.inputs.currentPagePoint
+		)
+
+		// todo: Extract into general hit test for arrows
+		if (this.editor.isShapeOfType<TLArrowShape>(shape, 'arrow')) {
+			// How should we handle multiple labels? Do shapes ever have multiple labels?
+			const labelGeometry = this.editor.getShapeGeometry<Group2d>(shape).children[1]
+			// Knowing what we know about arrows... if the shape has no text in its label,
+			// then the label geometry should not be there.
+			if (labelGeometry && pointInPolygon(pointInShapeSpace, labelGeometry.vertices)) {
+				return true
+			}
```

Comment:
nonblocking: I know this was in a previous PR, but it feels brittle to me, i wonder whether we could put the logic on the ArrowShapeUtil class, something like `arrowUtil.isPointWithinArrowLabelBounds(arrow, pointInShapeSpace)`?

Reply:
sounds good - will followup later 👍